src/interp/interp.h

```diff
@@ -443,10 +443,33 @@ class FreeList {
   std::vector<bool> is_free_;
 };
 
+class FreeRefList {
+ public:
+  using Index = size_t;
+
+  Index New(Ref);
+  void Delete(Index);
+
+  bool IsUsed(Index) const;
+
+  const Ref& Get(Index) const;
+  Ref& Get(Index);
+
+  Index size() const;  // 1 greater than the maximum index.
+
+ private:
+  static const size_t unusedStart = (SIZE_MAX >> 1) + 1;
```

Comment:
This probably warrants a comment.    Also, why not give this type `Index` ?  

Also, would it be more clear to the reader if we instead defined freeBit and use bitwise logic elsewhere rather than greater than / less then operations on this start number?   e.g. something like `bool isFree = myindex & FREE_BIT`.   I don't free strongly but it seems like when using magic bit packing it makes sense to use bitwise logic.

Reply:
Ok, from cpu perspective, there is no difference.